pkg/epp/handlers/server.go

```diff
@@ -190,8 +190,10 @@ func (s *StreamingServer) Process(srv extProcPb.ExternalProcessor_ProcessServer)
 			if len(requestID) == 0 {
 				requestID = uuid.NewString()
 				loggerTrace.Info("RequestID header is not found in the request, generated a request id")
-				reqCtx.Request.Headers[requtil.RequestIdHeaderKey] = requestID // update in headers so director can consume it
 			}
+			// Ensure the request ID, whether pre-existing or newly generated, is in the context's header map.
+			// This makes it available to all downstream logic (e.g,. the director).
```

Comment:
trying to understand if this is really a bug, in the case the header already existed, `HandleRequestHeaders` would have set it in the reqCtx.Request.Headers, right? I guess the only difference here is that we are unifying the header key as `requtil.RequestIdHeaderKey` since at line 187 the header could have had a different casing.

Reply:
> since at line 187 the header could have had a different casing

Let me look into this more. You may be onto something here. Perhaps there is a simpler solution / root cause for the behaviors I was seeing.